### PR TITLE
Document custom container images built for tmt tests

### PR DESCRIPTION
**Pull request by happz**
Finish the naming scheme, document it, and add a couple of images to improve the coverage of `prepare/install` and package manager tests (CentOS Stream 9, Fedora 39 and 40).

Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]
The only nitpick `FROM quay.io/fedora/fedora:39` and `FROM registry.fedoraproject.org/fedora:39` are both used, could be same FROM

File: tests/images.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+function build_container_image () {
+    if [ "$PROVISION_HOW" != "container" ]; then
+        rlLogInfo "Skipping the build of $1 container image because of PROVISION_HOW=$PROVISION_HOW"
+        return
+    fi
+
+    # Try several times to build the container
+    # https://github.com/teemtee/tmt/issues/2063
+    build="make -C $_MAKEFILE_DIR images-tests/tmt/tests/container/$1"
+    rlRun "rlWaitForCmd '$build' -m 5 -d 5 -t 3600" || rlDie "Unable to prepare the image"
```
[the remark this answers is not in the document]
Comment (happz) [reply tests/images.sh:28]: Hmm, I copied it from elsewhere, maybe it's indeed not needed. I can drop it and let it go.

[the remark this answers is not in the document]
Comment (happz) [reply tests/images.sh]: Dropped in 1899c427

**Review by janhavlin — APPROVED** [review]

**Review by therazix — APPROVED** [review]

**Comment by happz** [comment]
* `testing-farm:fedora-40-x86_64:extended-unit-tests` failed because of something that looks like a Fedora CoreOS bug: https://artifacts.dev.testing-farm.io/05950d0b-050e-414b-be2c-0556b6aedf26/work-extended-unit-testsjrcsy4w2/plans/features/extended-unit-tests/execute/data/guest/default-0/tests/unit/with-system-packages/extended-1/output.txt
* `testing-farm:fedora-40-x86_64:provision` is our good friend, timeout in `/tests/prepare/multihost`: https://artifacts.osci.redhat.com/testing-farm/8d27772b-33ca-4d06-8044-3076497135a8/work-virtual5gzlheeg/plans/provision/virtual/execute/data/guest/default-0/tests/prepare/multihost-21/output.txt
* `testing-farm:fedora-rawhide-x86_64:full` fails in `/tests/discover/distgit` which would be hopefully addressed somewhere in https://github.com/teemtee/tmt/issues/2987: https://artifacts.dev.testing-farm.io/0a80af4b-8f99-4901-a13d-2e1c89edc7c0/work-discover8p0nqpzs/plans/features/steps/discover/execute/data/guest/default-0/tests/discover/distgit-1/output.txt

Three approvals -> merging.

**Pull request merged**